testsupport/tiers/checks.go

```diff
@@ -48,333 +47,234 @@ var (
-		numberOfToolchainRoles(2),
-		numberOfToolchainRoleBindings(3),
-		numberOfToolchainServiceAccounts(1),
-		gitOpsServiceLabel(),
+	return func(nsType string) []namespaceObjectsCheck {
+		switch tierName {
+		case testtier:
+			return []namespaceObjectsCheck{}
+		case appstudio:
+			checks := []namespaceObjectsCheck{
+				limitRange(defaultCPULimit, "750Mi", "10m", "64Mi"),
+				appstudioServiceAccount(),
+				appstudioUserActionsRole(),
+				appstudioUserActionsRoleBinding(),
+				appstudioViewRoleBinding(),
+				toolchainSaReadRole(),
+				userSaReadRoleBinding(),
+				numberOfLimitRanges(1),
+				numberOfToolchainRoles(2),
+				numberOfToolchainRoleBindings(3),
+				numberOfToolchainServiceAccounts(1),
+				gitOpsServiceLabel(),
+			}
+			checks = append(checks, commonNetworkPolicyChecks...)
+			checks = append(checks, networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))
+			return checks
+		case base1ns:
+			checks := append(commonChecks,
+				limitRange(defaultCPULimit, "750Mi", "10m", "64Mi"),
+				rbacEditRoleBinding(),
+				rbacEditRole(),
+				crtadminPodsRoleBinding(),
+				crtadminViewRoleBinding(),
+				execPodsRole(),
+				numberOfToolchainRoles(2),
+				numberOfToolchainRoleBindings(4))
+			checks = append(checks, commonNetworkPolicyChecks...)
+			checks = append(checks, networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))
+			return checks
+		default:
+			checks := append(commonChecks,
+				limitRange(defaultCPULimit, "750Mi", "10m", "64Mi"),
+				rbacEditRoleBinding(),
+				rbacEditRole(),
+				crtadminPodsRoleBinding(),
+				crtadminViewRoleBinding(),
+				execPodsRole(),
+				numberOfToolchainRoles(2),
+				numberOfToolchainRoleBindings(4))
+			checks = append(checks, commonNetworkPolicyChecks...)
+			var otherNamespaceKind string
+			switch nsType {
+			case "dev":
+				otherNamespaceKind = "stage"
+			case "stage":
+				otherNamespaceKind = "dev"
+			}
+			checks = append(checks, networkPolicyAllowFromCRW(), networkPolicyAllowFromOtherNamespace(otherNamespaceKind), numberOfNetworkPolicies(7))
+			return checks
+		}
 	}
-
-	checks = append(checks, append(commonNetworkPolicyChecks(), networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))...)
-	return checks
 }
 
-func (a *appstudioTierChecks) GetExpectedTemplateRefs(hostAwait *wait.HostAwaitility) TemplateRefs {
-	templateRefs := GetTemplateRefs(hostAwait, a.tierName)
-	verifyNsTypes(hostAwait.T, a.tierName, templateRefs, "appstudio")
-	return templateRefs
+func getExpectedTemplateRefs(tierName string) func(hostAwait *wait.HostAwaitility) TemplateRefs {
+	return func(hostAwait *wait.HostAwaitility) TemplateRefs {
+		switch tierName {
+		case appstudio:
+			templateRefs := GetTemplateRefs(hostAwait, tierName)
+			verifyNsTypes(hostAwait.T, tierName, templateRefs, "appstudio")
+			return templateRefs
+		case base1ns:
+			templateRefs := GetTemplateRefs(hostAwait, tierName)
+			verifyNsTypes(hostAwait.T, tierName, templateRefs, "dev")
+			return templateRefs
+		default:
```

Comment:
Same comment about default here.

Reply:
refactored, but instead of returning an error, the actual NSTemplateTier is used to return the templateRefs